go/interpreter/lfvm/opcode_test.go

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package lfvm
+
+import (
+	"regexp"
+	"slices"
+	"testing"
+)
+
+func TestOpcode_String(t *testing.T) {
+	validName := regexp.MustCompile(`^0x00[0-9A-Fa-f]{2}$`)
+	for i := 0; i < 256; i++ {
+		op := OpCode(i)
+		if !validName.MatchString(op.String()) && op > NUM_OPCODES {
```

Comment:
This test only fails if the tested operation is bigger than `NUM_OPCODES` ?

Reply:
no, if the teste op is bigger than `NUM_OPCODES` and it is not printed in the fashion of `0x00XX`, which should be the case for all opcodes bigger than `NUM_OPCODES`

Reply:
Got it now - this test is checking that for everything that is not a defined instruction the hex value is printed. This is not what would be expected for the only test that checks the string.

Please fix the name of the test accordingly to state what this test is covering. Also, in this case, why `i` not starting with `NUM_OPCODES`?

Reply:
`in this case, why i not starting with NUM_OPCODES?` because I also intended to test that for all the "known" opcodes, a string not resembling `0x00XX` is provided.

Reply:
To avoid this kind of logic errors, I would try to make more clear what range is being tested:
```
if op is named { // where named is the defined opcodes only (including INVALID) 
    t.Skip()
}
// do the test for invalid opcodes. 
```

It happens to be that now all lfvm opcodes are < NUM_OPCODES, but this may change.